engines/director/util.cpp

```diff
@@ -1224,13 +1224,13 @@ Common::String formatStringForDump(const Common::String &str) {
 // Director 2.0-4.0 Mac, MacRoman encoding
 //
 // ................................
-//   !"«»“”#$%&'‘’()*+,-./012345678
-// 9:;<=>?@AÁÀÂÄÃÅaáàâäãåÆæBbCÇcçDd
-// EÉÈÊËeéèêëFfﬁﬂGgHhIÍÌÎÏı.íìîïJjK
-// kLlMmNÑnñOÓÒÔÖΩØoóòôöõøŒœPpQqRrS
-// sßTtUÚÙÛÜuúùûüVvWwXxYyÿŸZz[\]^_`
-// {|}~.†°¢£§•¶®©™´¨≠∞±≤≥¥µ∂∑∏π∫ªºΩ
-// ¿¡¬√ƒ≈Δ…–—÷◊⁄¤‹›‡·‚„‰��¯˘˙˚¸˝˛.
+//  "#()*+,/0123456789:;<=>?@ABCDEF
+// GHXZ^`jnpr}.ÅÉÖâôõùü¢§Æ∞≤¥∏∫ªºΩæ
+// øOY\_ekoqw~ÄÇÑáêöúû†£©Ø±≥µπ¿¡¬√ƒ
+// LN[aÜç®PQRSTU]fghiyz{|àëíìîï™´¨≠
+// ≈∆«»… À°ÃÕŒœ–—Vè“”‘’÷◊ÿŸ⁄€‹›ﬁﬂWñ
+// ‡·‚„‰ÂÊ$%Á!JMéóòËÈ&'-.ÍÎ∂∑ÏÌÓÔlm
+// �ÒÚÛÙKcIdbsuvtäåıã•ß¶wˆ˜¯˘˙˚¸˝˛.
```

Comment:
The transform used for the comments isn't quite right. The table is a mapping of character ordinal to sort order, not the other way around.

Example Python:
```
ax = [0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f,
    0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f,
    0x20, 0x22, 0x23, 0x28, 0x29, 0x2a, 0x2b, 0x2c, 0x2f, 0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36,
    0x37, 0x38, 0x39, 0x3a, 0x3b, 0x3c, 0x3d, 0x3e, 0x3f, 0x40, 0x41, 0x42, 0x43, 0x44, 0x45, 0x46,
    0x47, 0x48, 0x58, 0x5a, 0x5e, 0x60, 0x6a, 0x6e, 0x70, 0x72, 0x7d, 0x7f, 0x81, 0x83, 0x85, 0x89,
    0x99, 0x9b, 0x9d, 0x9f, 0xa2, 0xa4, 0xae, 0xb0, 0xb2, 0xb4, 0xb8, 0xba, 0xbb, 0xbc, 0xbd, 0xbe,
    0xbf, 0x4f, 0x59, 0x5c, 0x5f, 0x65, 0x6b, 0x6f, 0x71, 0x77, 0x7e, 0x80, 0x82, 0x84, 0x87, 0x90,
    0x9a, 0x9c, 0x9e, 0xa0, 0xa3, 0xa9, 0xaf, 0xb1, 0xb3, 0xb5, 0xb9, 0xc0, 0xc1, 0xc2, 0xc3, 0xc4,
    0x4c, 0x4e, 0x5b, 0x61, 0x86, 0x8d, 0xa8, 0x50, 0x51, 0x52, 0x53, 0x54, 0x55, 0x5d, 0x66, 0x67,
    0x68, 0x69, 0x79, 0x7a, 0x7b, 0x7c, 0x88, 0x91, 0x92, 0x93, 0x94, 0x95, 0xaa, 0xab, 0xac, 0xad,
    0xc5, 0xc6, 0xc7, 0xc8, 0xc9, 0xca, 0xcb, 0xa1, 0xcc, 0xcd, 0xce, 0xcf, 0xd0, 0xd1, 0x56, 0x8f,
    0xd2, 0xd3, 0xd4, 0xd5, 0xd6, 0xd7, 0xd8, 0xd9, 0xda, 0xdb, 0xdc, 0xdd, 0xde, 0xdf, 0x57, 0x96,
    0xe0, 0xe1, 0xe2, 0xe3, 0xe4, 0xe5, 0xe6, 0x24, 0x25, 0xe7, 0x21, 0x4a, 0x4d, 0x8e, 0x97, 0x98,
    0xe8, 0xe9, 0x26, 0x27, 0x2d, 0x2e, 0xea, 0xeb, 0xb6, 0xb7, 0xec, 0xed, 0xee, 0xef, 0x6c, 0x6d,
    0xf0, 0xf1, 0xf2, 0xf3, 0xf4, 0x4b, 0x63, 0x49, 0x64, 0x62, 0x73, 0x75, 0x76, 0x74, 0x8a, 0x8c,
    0xf5, 0x8b, 0xa5, 0xa7, 0xa6, 0x77, 0xf6, 0xf7, 0xf8, 0xf9, 0xfa, 0xfb, 0xfc, 0xfd, 0xfe, 0xff,]

bx = [0] * 256
for i, x in enumerate(ax):
    bx[x] = i

print(repr(bytes(bx).decode('macroman')))
# '\x00\x01\x02\x03\x04\x05\x06\x07\x08\t\n\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f \xa0!"«»“”#$%&\'‘’()*+,-./0123456789:;<=>?@AÁÀÂÄÃÅaáàâäãåÆæBbCÇcçDdEÉÈÊËeéèêëFfﬁﬂGgHhIÍÌÎÏı\x00íìîïJjKkLlMmNÑnñOÓÒÔÖÕØoóòôöõøŒœPpQqRrSsßTtUÚÙÛÜuúùûüVvWwXxYyÿŸZz[\\]^_`{|}~\x7f†°¢£§•¶®©™´¨≠∞±≤≥¥µ∂∑∏π∫ªºΩ¿¡¬√ƒ≈∆…–—÷◊⁄€‹›‡·‚„‰\uf8ffˆ˜¯˘˙˚¸˝˛ˇ'

Reply:
Oh fudge, that's right. I'll fix that up

Reply:
Yeah, this is what my perl script was doing

Reply:
Fixed